src/text/raw_text_writer.rs

```diff
@@ -373,20 +449,8 @@ impl<'a, W: Write> Writer for RawTextWriter<W> {
 
             write!(output, ":{:0>2}", datetime.second())?;
             //                   ^-- delimiting colon, formatted second
-            if value.precision == Precision::Second {
-                RawTextWriter::write_offset(output, offset_minutes)?;
-                return Ok(());
-            }
-
-            // We want the fractional seconds to show as many decimal places as are interesting.
-            // Here we format the string as nanoseconds and then eliminate any trailing zeros.
-            //TODO: The initial conversion to datetime will discard precision greater than nanoseconds.
-            //TODO: Eliminate this allocation.
-            let fractional_text = format!("{:0>9}", datetime.nanosecond());
-            let fractional_text = fractional_text.trim_end_matches('0');
+            value.fmt_fractional_seconds(&mut *output)?;
```

Comment:
🗺️ A more robust implementation of text fractional seconds serialization is now provided by `Timestamp` directly. It is `pub(crate)`, keeping it out of the public API for now.